tests/python_client/testcases/test_insert.py

```diff
@@ -1382,19 +1382,143 @@ def test_upsert_in_specific_partition(self):
                 3. upsert in the given partition
         expected: raise no exception
         """
+        # create a collection and 2 partitions
+        c_name = cf.gen_unique_str(pre_upsert)
+        collection_w = self.init_collection_wrap(name=c_name)
+        collection_w.create_partition("partition_1")
+        collection_w.create_partition("partition_2")
+        cf.insert_data(collection_w)
+        collection_w.create_index(ct.default_float_vec_field_name, default_index_params)
+        collection_w.load()
+
+        # check the ids which will be upserted is in partition _default
         upsert_nb = 10
+        expr = f"int64 >= 0 && int64 < {upsert_nb}"
+        res = collection_w.query(expr, [default_float_name], ["_default"])[0]
+        assert len(res) == upsert_nb
+
+        # upsert ids in partition _default
+        data, float_values = cf.gen_default_data_for_upsert(upsert_nb)
+        collection_w.upsert(data=data, partition_name="_default")
+
+        # check the result in partition _default and others
+        res0 = collection_w.query(expr, [default_float_name], ["_default"])[0]
+        assert [res0[i][default_float_name] for i in range(upsert_nb)] == float_values.to_list()
+        res1 = collection_w.query(expr, [default_float_name], ["partition_1"])[0]
+        assert res1 == []
```

Comment:
This check condition is not rigorous enough to make sure that the other partitions are not impacted. If it is empty now, this condition will be true too. And upsert is combined with delete operation, so maybe it will be deleted.